src/lib.rs

```diff
@@ -64,6 +64,15 @@ impl CkFrom<CK_BBOOL> for bool {
   }
 }
 
+fn str_from_blank_padded(field: &[CK_UTF8CHAR]) -> Result<&str, Error> {
+  let decoded_str = match std::str::from_utf8(field) {
```

Comment:
Looking at where this function gets used in the code, I don't think anybody would _really_ expect this not to convert from UTF-8. It is also not input coming from the user, so I think it would be reasonable to assume that they are indeed UTF-8 encoded strings.

Therefore, what about making this a `from_utf8_lossy` converted string? This way it's not a `Result` type, and in the rare occasion where this is not UTF-8, folks can probably live with `?` in their strings.

Reply:
That sounds good to me! I agree 100% about it being a reasonable expectation for these to always convert (and was concerned about the same thing re: vendors getting it wrong).

I'll switch to `from_utf8_lossy` 🙂.